pkgs/top-level/all-packages.nix

```diff
@@ -27543,7 +27543,7 @@ in
 
   taskopen = callPackage ../applications/misc/taskopen { };
 
-  tdesktop = qt5.callPackage ../applications/networking/instant-messengers/telegram/tdesktop {
+  tdesktop = libsForQt5.callPackage ../applications/networking/instant-messengers/telegram/tdesktop {
     inherit (xorg) libpthreadstubs libXdmcp;
```

Comment:
BTW, is there any difference between using `qt5.callPackage` and `libsForQt5.callPackage`?

Both produced the same .drv file

The former allows overriding. For example, to use modern `tdesktop` on ancient NixOS:  `(pkgs.qt515.override {stdenv = pkgs.gcc10Stdenv;}).callPackage ./tdesktop { }; `. This will not work with `libsForQt515`

Does `libsForQt5.callPackage` have any advantage over `qt5.callPackage` ?
Why all-packages.nix have more `libsForQt5.callPackage` than  `qt5.callPackage` ?

Reply:
I see that there are 2 namespaces for Qt's own libs and for 3rd party.
This is good.

My question is only about using `.callPackage` for cases like `tdesktop`, `cutter`, `gnuplot`, `yubikey-manager-qt`, `calibre`, `qbittorrent`, and other **apps**, not libs.

Reply:
> For example, to use modern tdesktop on ancient NixOS

We only support 21.05 and not ancient versions :)